### DISCUSSION TITLE
Workaround waiting Bpmn implementation

### DISCUSSION THREAD

**Discussion by holpit** [elsa-workflows/elsa-core#3735] [Q&A]
Hi,
Meanwhile that Elsa 3 + Bpmn will be released, which are the features of Bpmn2 that Elsa would support?
Gateways? Signal? Message? SubProcess? Escalation? and so on?

Thank you

**Comment by sfmskywalker** [comment]
Hi @holpit 

The goal is to implement the full specification except for a few exceptions. We'll be taking a close look at implementations such as Camunda to better understand how things work. 

To be clear, BPMN2 support will not ship at the same time as Elsa 3 does. It will be a separate module, and priority of when this project should start has not yet been determined.

**Reply by holpit** [reply]
Hi @sfmskywalker 
Sure, it's clear that Bmpn is a step after the release of version 3.0. 
We have tested Camunda Run 7 but the roadmap to 8.0 is not compatible whit our needs (Bpmn 2 + full intergrated and customizable engine)  and so Elsa is back to our comparisions.

Thank you

**Reply by sfmskywalker** [reply]
Out of curiosity, are you able to share some of these needs that 8.0 will not offer?

**Reply by holpit** [reply]
Mainly Camunda 7, and even more Camunda 8, are black-boxes and the management of UserTasks is too trivial compared to our needs (healthcare no-code platform). The embedded versions are practically deprecated.
We need to customize tasks and interact with systems external to the workflow engine via events.
On the other hand, our processes are created by domain experts in the BPMN 2 standard and this is a mandatory requirement and Camunda has great support for Bpmn and Dmn.
We are therefore evaluating alternatives
